### ISSUE TITLE
Notifications

### ISSUE THREAD

**Issue by DylanVanAssche** [iRail/harbour-berail#14]
Provide some notifications when your train is late or cancelled.

**Comment by rubdos** [comment]
I think a (very basic) UX implementation of this feature is needed before I can start working on #44: the user should be able to select some kind of "active route".

Two suggestions:
- add a property `active` (or `favorite` or something alike) to indicate which trip(s?) should generate notifications.
- Use the active trip from the `TripDetailPage` to indicate which trip is followed for notifications. Afaik, the `TripDetailPage` is not currently in use, but I might lag some versions behind :-)

**Comment by DylanVanAssche** [comment]
`TripDetailPage` is used for showing the stops of vehicles. You can reach it by going to Liveboard -> select a vehicle. I always wanted to expand the routes to show intermediate stops. However, at the time of writing it, the iRail API didn't expose such a thing. This feature is now tracked here: https://github.com/iRail/iRail/issues/348

We can use the current `TripDetailPage` as active route. Another option would be to pick the route that will depart first. So if you plan your trip 30 min in advance for example, the first listed route is the 'active' one.

**Comment by DylanVanAssche** [comment]
I don't have an iPhone or iWatch, but there's an option already for iOS and watch integration:
https://www.railer.be/

**Comment by rubdos** [comment]
> We can use the current TripDetailPage as active route.

I don't think it supports a connected (multi-vehicle) route though, I suppose a separate page would be needed.

> Another option would be to pick the route that will depart first. So if you plan your trip 30 min in advance for example, the first listed route is the 'active' one.

I don't like that option. It means that I have to search for routes with a modified time if I want to pick the route I want to take.

> I don't have an iPhone or iWatch, but there's an option already for iOS and watch integration:
https://www.railer.be/

They seem to show the "current" route as a separate page on the iPhone; I don't immediately find a "timeline"-like screenshot of the Apple Watch.

I'm primarily interested in doing Pebble Timeline first, because a "real" interactive Pebble experience requires both an App on the watch itself *and* a lot more support in berail.

---

Just to be clear: the reason I'm writing on this issue, is because there's a common UX element between #14 and #44: selecting the routes that interest the user to be notified about. I merely suppose the selection of this "active route" should be the same for plain Sailfish notification and the Timeline support.

**Comment by DylanVanAssche** [comment]
> > We can use the current TripDetailPage as active route.
> 
> I don't think it supports a connected (multi-vehicle) route though, I suppose a separate page would be needed.

True, needs some changes to show properly the transfers between vehicles, however some components of that page might be useful?



> > Another option would be to pick the route that will depart first. So if you plan your trip 30 min in advance for example, the first listed route is the 'active' one.
> 
> I don't like that option. It means that I have to search for routes with a modified time if I want to pick the route I want to take.
> 
> >

You're right, we should have something like `Watch this trip`.



> > I don't have an iPhone or iWatch, but there's an option already for iOS and watch integration:
> > https://www.railer.be/
> 
> They seem to show the "current" route as a separate page on the iPhone; I don't immediately find a "timeline"-like screenshot of the Apple Watch.
> 
> I'm primarily interested in doing Pebble Timeline first, because a "real" interactive Pebble experience requires both an App on the watch itself _and_ a lot more support in berail.

Timeline is fine for sure :) I linked to the Railer app for some inspiration maybe :)

**Comment by rubdos** [comment]
> > > We can use the current TripDetailPage as active route.
> > 
> > I don't think it supports a connected (multi-vehicle) route though, I suppose a separate page would be needed.
> 
> True, needs some changes to show properly the transfers between vehicles, however some components of that page might be useful?

Yes, that might be true. You're most familiar with your code base ;-)

> > > Another option would be to pick the route that will depart first. So if you plan your trip 30 min in advance for example, the first listed route is the 'active' one.
> > 
> > 
> > I don't like that option. It means that I have to search for routes with a modified time if I want to pick the route I want to take.
> > >
> 
> You're right, we should have something like `Watch this trip`.

My suggestion: long-press, and a favorite star appears. Single tap is already used to view details, and I like that :-)

> > > I don't have an iPhone or iWatch, but there's an option already for iOS and watch integration:
> > > https://www.railer.be/
> > 
> > 
> > They seem to show the "current" route as a separate page on the iPhone; I don't immediately find a "timeline"-like screenshot of the Apple Watch.
> > I'm primarily interested in doing Pebble Timeline first, because a "real" interactive Pebble experience requires both an App on the watch itself _and_ a lot more support in berail.
> 
> Timeline is fine for sure :) I linked to the Railer app for some inspiration maybe :)

Thanks for that, I appreciate the discussion!

**Comment by DylanVanAssche** [comment]
> Yes, that might be true. You're most familiar with your code base ;-)

AFAIK, the component that creates a vehicle trip, must be generated multiple times (for each vehicle in the trip) on the page. The header can be upgraded to show trip information like departure station and arrival station or other information.

**Comment by rubdos** [comment]
So, which approach would you take to enable/disable notifications? If you take a long-press approach to add to a "watch list", one could follow multiple routes at the same time -- which I often try to do when e.g. taking Hemiksem - Mouterij (there's a 4 minute "connection" in Antwerp...).

**Comment by DylanVanAssche** [comment]
A watch list would be ideal I guess. Something like `Planned trips` or `Watch list`.
I suppose that trips that are in the past are removed from the list since no notifications will appear for past trips.

**Comment by rubdos** [comment]
> A watch list would be ideal I guess. Something like `Planned trips` or `Watch list`.

Unless you have a better/more interesting idea! I'll give it a shot in about half an hour.

> I suppose that trips that are in the past are removed from the list since no notifications will appear for past trips.

Absolutely. Train arrived/cancelled/... should trigger deletion. It's probably okay to keep it in-RAM too; no need for persistence I'd think.

**Comment by DylanVanAssche** [comment]
> Unless you have a better/more interesting idea! I'll give it a shot in about half an hour.

Awesome!

> Absolutely. Train arrived/cancelled/... should trigger deletion. It's probably okay to keep it in-RAM too; no need for persistence I'd think.

Sure for this use case that's more than sufficient :)
